common/build/eslint-config-fluid/flat.mts

```diff
@@ -154,203 +95,540 @@ const restrictedImportPaths = [
 	},
 };
-recommended.push(internalModulesConfig);
-strict.push(internalModulesConfig);
-minimalDeprecated.push(internalModulesConfig);
-
-// ESLint 9 upgrade: Disable new react-hooks rules that were introduced in eslint-plugin-react-hooks 7.0.
-// These rules have violations across the codebase that need to be addressed incrementally.
-// TODO: Fix violations and enable these rules as errors.
-const reactHooksEslint9Upgrade = {
+
+/**
+ * React/JSX file configuration.
+ */
+const reactConfig: Linter.Config = {
 	files: ["**/*.jsx", "**/*.tsx"],
+	plugins: { "react": reactPlugin, "react-hooks": reactHooksPlugin },
+	settings: { react: { version: "detect" } },
 	rules: {
-		// react-hooks/immutability: Warns about mutating variables during render
-		// https://github.com/facebook/react/pull/29456
+		...reactPlugin.configs.recommended.rules,
+		...reactHooksPlugin.configs.recommended.rules,
+		// ESLint 9: New react-hooks rules set to warn for incremental adoption
 		"react-hooks/immutability": "warn",
-
-		// react-hooks/refs: Warns about reading refs during render (before useEffect)
-		// https://github.com/facebook/react/pull/29516
 		"react-hooks/refs": "warn",
-
-		// react-hooks/set-state-in-effect: Warns about calling setState synchronously in effects
-		// https://github.com/facebook/react/pull/30224
+		"react-hooks/rules-of-hooks": "warn",
 		"react-hooks/set-state-in-effect": "warn",
-
-		// react-hooks/static-components: Warns about creating components during render
-		// https://github.com/facebook/react/pull/30239
 		"react-hooks/static-components": "warn",
+		"unicorn/consistent-function-scoping": "off",
 	},
 };
-recommended.push(reactHooksEslint9Upgrade);
-strict.push(reactHooksEslint9Upgrade);
-minimalDeprecated.push(reactHooksEslint9Upgrade);
 
-// CommonJS files (.cts, .cjs) can use __dirname and require, which are valid in CommonJS
-const cjsFileConfig = {
-	files: ["**/*.cts", "**/*.cjs"],
+/**
+ * Test file configuration.
+ */
+const testConfig: Linter.Config = {
+	files: testFilePatterns,
+	languageOptions: {
+		parserOptions: {
+			projectService: false,
+			project: ["./tsconfig.json", "./src/test/tsconfig.json"],
+		},
+	},
 	rules: {
-		"unicorn/prefer-module": "off", // __dirname and require are valid in CommonJS
+		"@typescript-eslint/no-invalid-this": "off",
+		"@typescript-eslint/unbound-method": "off",
+		"@typescript-eslint/consistent-type-exports": "off",
+		"@typescript-eslint/consistent-type-imports": "off",
+		"@typescript-eslint/no-restricted-imports": ["error", { paths: restrictedImportPaths }],
+		"import-x/no-nodejs-modules": "off",
+		"import-x/no-deprecated": "off",
+		"import-x/no-internal-modules": [
+			"error",
+			{ allow: ["@fluid*/*/test*", "@fluid*/*/internal/test*", ...permittedImports] },
+		],
+		"import-x/no-extraneous-dependencies": ["error", { devDependencies: true }],
+		"unicorn/consistent-function-scoping": "off",
+		"unicorn/prefer-module": "off",
```

Comment:
I think it would be clearer if rules are not inlined in a config - declare the ruleset separately, then we can use the same pattern everywhere where the configs just assemble earlier delcared things.